aspnetcore/performance/caching/middleware.md

```diff
@@ -5,31 +5,30 @@ description: Learn how to configure and use Response Caching Middleware in ASP.N
 monikerRange: '>= aspnetcore-3.1'
 ms.author: riande
 ms.custom: mvc
-ms.date: 12/07/2021
+ms.date: 1/1/2022
 no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
 uid: performance/caching/middleware
 ---
 # Response Caching Middleware in ASP.NET Core
 
-By [John Luo](https://github.com/JunTaoLuo)
+By [John Luo](https://github.com/JunTaoLuo) and [Rick Anderson](https://twitter.com/RickAndMSFT)
 
 ::: moniker range=">= aspnetcore-6.0"
 
-This article explains how to configure Response Caching Middleware in an ASP.NET Core app. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the [`[ResponseCache]`](xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute) attribute, see [Response Caching](xref:performance/caching/response).
-
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/caching/middleware/samples) ([how to download](xref:index#how-to-download-a-sample))
+This article explains how to configure [Response Caching Middleware](https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs) in an ASP.NET Core app. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the [`[ResponseCache]`](xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute) attribute, see [Response Caching](xref:performance/caching/response).
 
-## Configuration
-
-Response Caching Middleware is implicitly available for ASP.NET Core apps via the shared framework.
+The Response caching middleware:
 
-In `Program.cs`, add the Response Caching Middleware to the service collection:
+* Implements the standard HTTP caching semantics, that is it caches based on HTTP cache headers like proxies do.
+* Is typically not beneficial for UI apps such as Razor Pages. Output caching is being considered for the next version of ASP.NET Core, which will benefit UI apps. With output caching, the app, not the browser, decides what should be cached. For more information, see [this GitHub issue](https://github.com/dotnet/aspnetcore/issues/27387).
+* May be beneficial for public GET or HEAD API requests from clients where the [Conditions for caching](#cfc) are met.
```

Comment:
@adityamandaleeka @halter73 please review these two lines. Feel free to look at the rest of the doc.

Reply:
> With output caching, the app, not the browser, decides what should be cached.

It would imply that this is the browser that decides what should be cached. Though even with Response Caching this is not the case, it's "response headers" based. I would suggest:

"With output caching, configuration decides what should be cached independently of HTTP headers."